src/truncate.jl

```diff
@@ -49,33 +49,94 @@ insupport(d::Truncated{D,Union{Discrete,Continuous}}, x::Real) where {D<:Univari
 
 quantile(d::Truncated, p::Real) = quantile(d.untruncated, d.lcdf + p * d.tp)
 
-for f in [:pdf, :logpdf, :cdf, :logcdf, :ccdf, :logccdf]
-    @eval ($f)(d::Truncated, x::Int) = ($f)(d, float(x))
+function _pdf(d::Truncated, x::T) where {T<:Real}
+    if d.lower <= x <= d.upper
+        pdf(d.untruncated, x) / d.tp
+    else
+        zero(T)
+    end
+end
+
+function pdf(d::Truncated{D}, x::T) where {D<:ContinuousUnivariateDistribution, T<:Real}
```

Comment:
This is very slightly shorter. You could also switch to the short function form for one-liners.
```suggestion
function pdf(d::Truncated{<:ContinuousUnivariateDistribution}, x::T) where {T<:Real}
```